File: giphy/src/main/java/com/valgoodchat/giphy/data/DataManager.java
package com.valgoodchat.giphy.data;

import com.valgoodchat.giphy.model.GiphyModel;
import com.valgoodchat.giphy.service.GiphyService;
import com.valgoodchat.giphy.service.ServiceFactory;

import rx.Observable;

/**
 * Class to handle all the operations performed with Giphy API
 */
public class DataManager {
    private static final String TAG = "DataManager";

    public static Observable<GiphyModel> searchGiphyByKeyword(String keyword,
                                                              int limit,
                                                              String pos,
                                                              String locale) {
        GiphyService service =
            ServiceFactory.createRetrofitService(GiphyService.class, GiphyService.SERVICE_ENDPOINT);

        return service.searchGif(keyword, pos, String.valueOf(limit), locale);
    }

    public static Observable<GiphyModel> getTrending(int limit, String pos) {
        GiphyService service =
                ServiceFactory.createRetrofitService(GiphyService.class, GiphyService.SERVICE_ENDPOINT);

        return service.trending(String.valueOf(limit), pos);
    }
}

File: giphy/src/main/java/com/valgoodchat/giphy/data/DataManager.kt
package com.valgoodchat.giphy.data

import com.valgoodchat.giphy.model.GiphyModel
import com.valgoodchat.giphy.service.GiphyService
import com.valgoodchat.giphy.service.ServiceFactory

import rx.Observable

/**
 * Class to handle all the operations performed with Giphy API
 */
object DataManager {
    fun searchGiphyByKeyword(keyword: String,
                             limit: Int,
                             pos: String,
                             locale: String): Observable<GiphyModel> {
        val service = ServiceFactory.createRetrofitService(GiphyService::class.java)

        return service.searchGif(keyword, pos, limit.toString(), locale)
    }

    fun getTrending(limit: Int, pos: String): Observable<GiphyModel> {
        val service = ServiceFactory.createRetrofitService(GiphyService::class.java)

        return service.trending(limit.toString(), pos)
    }
}
